packages/smart-contracts/src/contracts/ERC20Proxy.sol

```diff
@@ -0,0 +1,37 @@
+pragma solidity ^0.5.0;
+
+import "openzeppelin-solidity/contracts/token/ERC20/ERC20.sol";
+
+/**
+ * @title ERC20Proxy
+ * @notice This contract performs a ERC20 token transfer and store a reference
+  */
+contract ERC20Proxy {
+  // Event to declare a transfer with a reference
+  event TransferWithReference(address tokenAddress, address indexed to, uint256 amount, bytes indexed transferReference);
```

Comment:
in ETH we call it  "payment reference", not "transfer reference", it will be confusing